File: hw/vizcaino/u3/hw21FASPyJs/hw21JavaScript/view/animalFarmSystem.js
import { Cow } from "../model/cow.js";
import { Pig } from "../model/pig.js";
import { Product } from "../model/product.js";
import { SlaughterHouse } from "../model/slaughterHouse.js";

const slaughterHouse = new SlaughterHouse(
    1,
    "Central Slaughter House",
    "Quito",
    "0999999999"
);

const product = new Product(
    1,
    "Milk",
    "Liters",
    0
);

const cuts = [];

const pig = new Pig(
    120,
    35,
    1,
    "Yorkshire",
    new Date(2024, 0, 1),
    130,
    slaughterHouse,
    null,
    cuts
);

const cow = new Cow(
    true,
    62,
    2,
    "Holstein",
    new Date(2022, 5, 1),
    550,
    slaughterHouse,
    product,
    cuts
);

console.log("Hello World!");

console.log(pig.toString());
console.log(cow.toString());

console.log(pig.cut());

cow.measureQuantity(
    "Liters",
    15.5
);

console.log(cow.produce());

console.log(cow.cut());

File: hw/vizcaino/u3/hw21FASPyJs/hw21Python/view/animal_farm_system.py
from datetime import date

from model.cow import Cow
from model.pig import Pig
from model.product import Product
from model.slaughter_house import SlaughterHouse


def main():
    slaughter_house = SlaughterHouse(
        1,
        "Central Slaughter House",
        "Quito",
        "0999999999",
    )

    product = Product(
        1,
        "Milk",
        "Liters",
        0.0,
    )

    cuts = []

    pig = Pig(
        120.0,
        35.0,
        1,
        "Yorkshire",
        date(2024, 1, 1),
        130.0,
        slaughter_house,
        None,
        cuts,
    )

    cow = Cow(
        True,
        62.0,
        2,
        "Holstein",
        date(2022, 6, 1),
        550.0,
        slaughter_house,
        product,
        cuts,
    )

    print(pig)
    print(cow)

    print(pig.cut())

    cow.measure_quantity(
        "Liters",
        15.5,
    )

    print(cow.produce())
    print(cow.cut())


if __name__ == "__main__":
    main()